runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -0,0 +1,1350 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "codegen/CodeGenerator.hpp"
+#include "compile/ResolvedMethod.hpp"
+#include "env/VMAccessCriticalSection.hpp"
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+#include "il/SymbolReference.hpp"
+#include "optimizer/VectorAPIExpansion.hpp"
+
+
+const char *
+TR_VectorAPIExpansion::optDetailString() const throw()
+   {
+   return "O^O VECTOR API EXPANSION: ";
+   }
+
+#define OPT_DETAILS_VECTOR "O^O VECTOR API: "
+
+
+int32_t
+TR_VectorAPIExpansion::perform()
+   {
+   // TODO: add command line options after openj9 PR is merged
+   static bool disableVectorAPIExpansion = feGetEnv("TR_disableVectorAPIExpansion") ? true : false;
+   static bool traceVectorAPIExpansion = feGetEnv("TR_traceVectorAPIExpansion") ? true : false;
+
+   _trace = traceVectorAPIExpansion;
+   
+   if (!disableVectorAPIExpansion &&
+       !TR::Compiler->om.canGenerateArraylets() &&
+       findVectorMethods())
```

Comment:
Do we have guarded this opt to run only in java version from where methods from Vector JEPs are introduced. For example we do not need to run this optimization in Java 8 and Java 11. `findVectorMethods()` call would start walking tree tops to find if the method contains a call to the recognized vector API methods.